ocaml/idl/datamodel_vm.ml

```diff
@@ -1514,6 +1514,19 @@ let set_appliance =
       ]
     ~allowed_roles:_R_POOL_OP ()
 
+let set_groups =
+  call ~name:"set_groups" ~in_product_since:rel_boston
```

Comment:
This method would add the VM to multiple A/A groups but currently this is limit to one A/A group. Do we really want to support adding a VM to multiple A/A groups in one call? This already creates two failure modes: the set is empty or contains more than one element. An alternative would be to add a VM to exactly one A/A group and you would need multiple calls to add it to more than one. I am also skeptical about the name; "set" is a not carrying a lot of meaning and  would consider "add" or "inject" stronger. "VM.set_groups" carries little meaning for me.

Reply:
https://info.citrite.net/display/xenserver/Design+of+VM+anti-affinity+support+on+Toolstack#:~:text=Description-,set_groups,-VM%20Admin
We have already discussed the API in the design. In most cases, the VM only belongs to one A/A group, so the "set" is OK. The empty value for the set is not a failure, it means remove the VM from groups.

Reply:
I understand the methods like "VM.add_group" or "VM.join_group" can be added in the future if the VM group will be re-used by other features.
At the moment, it would be easier to create a minimum set APIs.

Reply:
I think the idea of this design is for simplicity, otherwise without a `setter`, we will have to provide an `adder`, `remover` and for the error: "vm can only belong to one anti-affinity group", it might also arise with an `adder` if it is called multiple times.